### PR TITLE
Improve Structure and Naming

### PR DESCRIPTION
**Pull request by suho**
## What happened 👀

- Updated structure and naming to follow conventions and best practices.

## Insight 📝

- When rethink on different use cases (one path but multiple operations). It's better to improve the way we separate files and file naming.
- Also add the folder `request_bodies` to support reuse later.

## Proof Of Work 📹

Screenshot 2023-06-26 at 12 06 14

### CODE REVIEW

File: docs/openapi/paths/registrations.yml [hunk 1]
```diff
@@ -6,18 +6,17 @@ post:
...
-
     The backend system will send an email to the user to create a new account on the website.
+
   requestBody:
     required: true
     content:
       application/json:
         schema:
-          $ref: '../../schemas.yml#/requests_registrations_post_create'
+          $ref: '../schemas.yml#/requests_registrations_post_create'
```
Comment (hoangmirs) [inline docs/openapi/paths/registrations.yml:16]: Can we use `post` or `create` only?
For me, I prefer `requests_registrations_post` as we are using the method name in the top.

Comment (suho) [reply docs/openapi/paths/registrations.yml:16]: Good catch! Let me refactor in schemas folder

Comment (suho) [reply docs/openapi/paths/registrations.yml]: Improve in this commit: https://github.com/nimblehq/nimble-survey-web/pull/95/commits/099fd28cdb43f0f480d5aa71c229a73d81b669db

File: docs/openapi/schemas.yml [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
 
-responses_oauth_post_token:
-  $ref: "schemas/responses/oauth/post_token.yml"
+responses_oauth_token:
+  $ref: "schemas/responses/oauth/token.yml"
 
-responses_surveys_get_survey_list:
-  $ref: "schemas/responses/surveys/get_survey_list.yml"
+responses_surveys:
+  $ref: "schemas/responses/surveys.yml"
```
Comment (hoangmirs) [inline docs/openapi/schemas.yml:22]: After rechecking, I think we can add the RESTful action name here:
```suggestion
responses_surveys_index:
  $ref: "schemas/responses/surveys/index.yml"
```
Then later, we can will have another for resource detail:

```yml
responses_surveys_index:
  $ref: "schemas/responses/surveys/show.yml"
```

Same for `update`/`delete`
What do you think?

Comment (suho) [reply docs/openapi/schemas.yml:22]: @hoangmirs That's a good raise. I was thinking about following the RESTful action names. However when checking with OpenAPI, it supports `get`, `post`, `put`, `patch`, `delete`, `head`, `options`, and `trace` operations. So I prefer to use operations for easy-to-find and expandable.

Moreover, for surveys show (endpoints: `surveys/{id}`), it's defined with another path, so we can do this

For examples
```yml
# openapi.yml

paths:
  /surveys:
    $ref: "paths/surveys.yml"

  /surveys/{id}:
    $ref: "paths/surveys_id.yml"
```
```yml
# paths/surveys.yml

get:
  $ref: "paths/surveys/get.yml"

post
  $ref: "paths/surveys/post.yml"
```
```yml
# paths/surveys_id.yml
get:
  ...
```

**Review by hoangmirs — APPROVED** [review]

**Review by olivierobert — APPROVED** [review]

**Pull request merged**